libvast/src/format/zeek.cpp

```diff
@@ -554,64 +456,159 @@ writer::writer(path dir) {
 }
 
 writer::~writer() {
-  if (streams_.empty())
-    return;
-  std::ostringstream ss;
-  ss << "#close" << separator << time_factory{} << '\n';
-  auto footer = ss.str();
-  for (auto& pair : streams_)
-    if (pair.second)
-      *pair.second << footer;
+  // nop
 }
 
-expected<void> writer::write(const event& e) {
-  if (!caf::holds_alternative<record_type>(e.type()))
-    return make_error(ec::format_error, "cannot process non-record events");
-  std::ostream* os = nullptr;
+namespace {
+
+template <class Iterator>
+class zeek_printer : public printer<zeek_printer<Iterator>> {
+public:
+  using attribute = view<data>;
+
+  bool operator()(Iterator& out, view<caf::none_t>) const {
+    *out++ = unset_field;
+    return true;
+  }
+
+  template <class T>
+  bool operator()(Iterator& out, const T& x) const {
+    if constexpr (detail::is_any_v<T, view<vector>, view<set>>) {
+      if (x.empty()) {
+        for (auto c : std::string_view(empty_field))
+          *out++ = c;
+        return true;
+      }
+      auto i = x.begin();
+      if (!print(out, *i))
+        return false;
+      for (++i; i != x.end(); ++i) {
+        *out++ = set_separator;
+        if (!print(out, *i))
+          return false;
+      }
+      return true;
+    } else if constexpr (std::is_same_v<T, view<map>>) {
+      VAST_ERROR(this, "cannot print maps in Zeek format");
```

Comment:
This seems to suggest that VAST is the problem. But it's Zeek TSV format that cannot handle maps. Therefore, I'd rephrase this as follows:

```
      VAST_ERROR(this, "Zeek TSV format does not support maps");
```